compiler/test/stdlib/immutableset.test.gr

```diff
@@ -0,0 +1,199 @@
+import ImmutableSet from "immutableset"
+import List from "list"
+import Array from "array"
+
+// Data types used in multiple tests
+enum Resource {
+  Grain,
+  Sheep,
+  Brick,
+  Wood,
+}
+record ResourceData {
+  name: String,
+  emoji: String,
+}
+
+// ImmutableSet.isEmpty
+
+let mut e = ImmutableSet.empty
+
+assert ImmutableSet.isEmpty(e)
+e = ImmutableSet.remove("🌾", e)
+assert ImmutableSet.isEmpty(e)
+let newE = ImmutableSet.add("🌾", e)
+assert !ImmutableSet.isEmpty(newE)
+assert ImmutableSet.isEmpty(e)
+
+// ImmutableSet.size
+
+let s = ImmutableSet.fromList(["🌾", "🐑", "🧱"])
+
+assert ImmutableSet.size(s) == 3
+
+// ImmutableSet.contains
+
+let h = ImmutableSet.fromList(["🌾", "🐑", "🧱"])
+
+assert ImmutableSet.contains("🌾", h)
+assert ImmutableSet.contains("🐑", h)
+assert ImmutableSet.contains("🧱", h)
+assert !ImmutableSet.contains("🌳", h)
+
+// ImmutableSet.add
+
+let mut vars = ImmutableSet.empty
+
+vars = ImmutableSet.add(Grain, vars)
+vars = ImmutableSet.add(Sheep, vars)
+vars = ImmutableSet.add(Grain, vars)
+
+assert ImmutableSet.size(vars) == 2
+
+let mut recs = ImmutableSet.empty
+
+recs = ImmutableSet.add({ name: "Grain", emoji: "🌾" }, recs)
+recs = ImmutableSet.add({ name: "Sheep", emoji: "🐑" }, recs)
+recs = ImmutableSet.add({ name: "Brick", emoji: "🧱" }, recs)
+recs = ImmutableSet.add({ name: "Grain", emoji: "🌾" }, recs)
+recs = ImmutableSet.add({ name: "Sheep", emoji: "🐑" }, recs)
+recs = ImmutableSet.add({ name: "Brick", emoji: "🧱" }, recs)
+
+assert ImmutableSet.size(recs) == 3
+
+// ImmutableSet.remove
+
+let mut r = ImmutableSet.fromList(["🌾", "🐑", "🧱"])
+
+assert ImmutableSet.size(r) == 3
+
+r = ImmutableSet.remove("🌾", r)
+
+assert ImmutableSet.size(r) == 2
+assert !ImmutableSet.contains("🌾", r)
+
+r = ImmutableSet.remove("🐑", r)
+r = ImmutableSet.remove("🧱", r)
+
+assert ImmutableSet.isEmpty(r)
+
+// ImmutableSet.filter
+
+let makeTestSet = () => ImmutableSet.fromList([Grain, Sheep, Brick])
+
+let emptySet = ImmutableSet.empty
```

Comment:
```suggestion
```
Can you remove this `emptySet` and replace the usages with `ImmutableSet.empty` directly?